picky/src/Components/Basic/Login.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import TextField from './TextField';
+import Products from '../Pages/Products';
+
+class Login extends Component {
+  constructor(props) {
+    super(props);
+      this.state = {
+        username : '',
+        password : '',
+        isUserNew : false,
+    };
+    this.handleData = this.handleData.bind(this);
+    this.getData = this.getData.bind(this);
+    this.signUp = this.signUp.bind(this);
+  }
+
+  handleData(field) {
+      return (data) => {
+        const x = Object.assign({}, this.state);
+        x[field] = data.target.value;
+        this.setState(x);
+        console.log(this.state);
+      };
+  }
+
+  getData() {
+    const username = this.state.username;
+    const password = this.state.password;
+    console.log("username is " + username + " and password is " + password);
+  }
+
+  signUp() {
+    const isUserNew = Object.assign({}, this.state, { isUserNew: true });
```

Comment:
Make sure you polyfill this if you're worried about cross-browser compatibility.

See [MDN](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) for more. The browser support is at the bottom.